crates/storage/errors/src/db.rs

```diff
@@ -1,57 +1,97 @@
-use std::{fmt::Display, str::FromStr};
-use thiserror::Error;
+use core::{
+    fmt::{Display, Result},
+    str::FromStr,
+};
+
+extern crate alloc;
+use alloc::boxed::Box;
+
+#[cfg(feature = "std")]
+use std::{error::Error, fmt};
```

Comment:
I prefer using the fully qualified path where necessary and not feature gate the `use` import

Reply:
Is this from a readability perspestive?

Reply:
yes and reduces cfgs